core/tasks.py

```diff
@@ -498,25 +527,33 @@ def task():
                 env["custom"] = conf["viur.tasks.customEnvironmentHandler"][0]()
 
             # Create task description
-            task = {
-                "app_engine_http_request": {  # Specify the type of request.
-                    "body": json.dumps(preprocessJsonObject((command, (funcPath, args, kwargs, env)))).encode("UTF-8"),
-                    "http_method": tasks_v2.HttpMethod.POST,
-                    "relative_uri": taskargs["url"]
-                }
-            }
-
-            # Set a schedule time in case countdown was set.
+            task = tasks_v2.Task(
+                app_engine_http_request=tasks_v2.AppEngineHttpRequest(
+                    body=json.dumps(preprocessJsonObject((command, (funcPath, args, kwargs, env)))).encode("UTF-8"),
+                    http_method=tasks_v2.HttpMethod.POST,
+                    relative_uri=taskargs["url"],
+                    app_engine_routing=tasks_v2.AppEngineRouting(
+                        version=taskargs.get("target_version") or os.getenv("GAE_VERSION"),
```

Comment:
```suggestion
                        version=taskargs.get("target_version", conf["viur.instance.app_version"]),
```
We can use here `conf["viur.instance.app_version"]` too.